src/test/java/com/simon/smile/user/address/AddressServiceTest.java

```diff
@@ -0,0 +1,180 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class AddressServiceTest {
+    private static final String ERROR_ADDRESS_NOT_FOUND = "Not found the address with ID: 1";
+
+    @Mock
+    private AddressRepository addressRepository;
+    @InjectMocks
+    private AddressService addressService;
+    private final AppUser appUser = new AppUser()
+            .setId(1)
+            .setUsername("test")
+            .setEmail("test@exmaple.com")
+            .setEnabled(true)
+            .setRoles("ROLE_USER");
+    private Address address;
+
+    private List<Address> addressList;
+
+    @BeforeEach
+    void setUp() {
+        address = new Address()
+                .setId(1)
+                .setDefault(true)
+                .setOwner(appUser)
+                .setPhone("13012345678")
+                .setAddress("test address");
+
+        addressList = List.of(
+                address,
+                new Address()
+                        .setId(2)
+                        .setAddress("test address 2")
+                        .setPhone("13012345678")
+                        .setOwner(appUser)
+                        .setDefault(false)
+        );
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    @DisplayName("Verify create address success")
+    void validCreateSuccess() {
+
+        given(addressRepository.save(any(Address.class))).willReturn(address);
+
+        Address createdAddress = addressService.create(address);
+
+        assertThat(createdAddress.getAddress()).isEqualTo(address.getAddress());
+        assertThat(createdAddress.getPhone()).isEqualTo(address.getPhone());
+        assertThat(createdAddress.getOwner()).isEqualTo(address.getOwner());
+        assertThat(createdAddress.isDefault()).isEqualTo(address.isDefault());
+        verify(addressRepository, times(1)).save(any(Address.class));
+    }
+
+    @Test
+    @DisplayName("Verify delete address success")
+    void validDeleteSuccess() {
+        given(addressRepository.findById(1)).willReturn(Optional.of(address));
+
+        doNothing().when(addressRepository).deleteById(anyInt());
+
+        addressService.delete(1);
+
+        verify(addressRepository, times(1)).findById(anyInt());
+        verify(addressRepository, times(1)).deleteById(anyInt());
+    }
+
+    @Test
+    @DisplayName("Verify delete address error when id not exist")
+    void validDeleteErrorWhenIDNotExist() {
+        given(addressRepository.findById(1)).willThrow(new ObjectNotFoundException(ERROR_ADDRESS_NOT_FOUND));
+
+        Throwable throwable = catchThrowable(() -> addressService.delete(1));
+
+        assertThat(throwable).isInstanceOf(ObjectNotFoundException.class);
```

Comment:
(sonar-qube) Join these multiple assertions subject to one assertion chain.
---
Consecutive AssertJ `assertThat` statements should be chained.

Reply:
done